src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> temporaryNode;
```

Comment:
You should have only the `head` and `tail` nodes of the class
It is better to create a new Node in the methods
```suggestion
```

Reply:
Why is this solution better than the one I wrote?